offchain-modules/packages/x/src/handlers/eth.ts

```diff
@@ -320,7 +375,10 @@ export class EthHandler {
     await this.init();
     this.watchNewBlock();
     this.watchLockEvents();
-    this.watchUnlockEvents();
+    if (this.role !== 'collector') {
```

Comment:
I think all roles(watcher, collector and verifier) can share the same watch logic